src/pages/home/report/ReportActionCompose.js

```diff
@@ -159,1008 +172,974 @@ const getMaxArrowIndex = (numRows, isAutoSuggestionPickerLarge) => {
-                            shouldIncludeReportRecipientLocalTimeHeight={shouldShowReportRecipientLocalTime}
-                        />
-                    </ArrowKeyFocusManager>
-                )}
-                {!_.isEmpty(this.state.suggestedMentions) && this.state.shouldShowMentionSuggestionMenu && (
-                    <ArrowKeyFocusManager
-                        focusedIndex={this.state.highlightedMentionIndex}
-                        maxIndex={getMaxArrowIndex(this.state.suggestedMentions.length, this.state.isAutoSuggestionPickerLarge)}
-                        shouldExcludeTextAreaNodes={false}
-                        onFocusedIndexChanged={(index) => this.setState({highlightedMentionIndex: index})}
-                    >
-                        <MentionSuggestions
-                            onClose={() => this.setState({suggestedMentions: []})}
-                            highlightedMentionIndex={this.state.highlightedMentionIndex}
-                            mentions={this.state.suggestedMentions}
-                            comment={this.state.value}
-                            updateComment={(newComment) => this.setState({value: newComment})}
-                            colonIndex={this.state.colonIndex}
-                            prefix={this.state.mentionPrefix}
-                            onSelect={this.insertSelectedMention}
-                            isComposerFullSize={this.props.isComposerFullSize}
-                            isMentionPickerLarge={this.state.isAutoSuggestionPickerLarge}
-                            composerHeight={this.state.composerHeight}
-                            shouldIncludeReportRecipientLocalTimeHeight={shouldShowReportRecipientLocalTime}
-                        />
-                    </ArrowKeyFocusManager>
-                )}
-            </View>
-        );
-    }
+                </View>
+                <View
+                    style={[
+                        styles.flexRow,
+                        styles.justifyContentBetween,
+                        styles.alignItemsCenter,
+                        (!props.isSmallScreenWidth || (props.isSmallScreenWidth && !props.network.isOffline)) && styles.chatItemComposeSecondaryRow,
+                    ]}
+                >
+                    {!props.isSmallScreenWidth && <OfflineIndicator containerStyles={[styles.chatItemComposeSecondaryRow]} />}
+                    <ReportTypingIndicator reportID={props.reportID} />
+                    <ExceededCommentLength
+                        comment={comment.current}
+                        onExceededMaxCommentLength={setExceededMaxCommentLength}
+                    />
+                </View>
+            </OfflineWithFeedback>
+            {isDraggingOver && <ReportDropUI />}
+            {!_.isEmpty(suggestionValues.suggestedEmojis) && suggestionValues.shouldShowEmojiSuggestionMenu && (
+                <ArrowKeyFocusManager
```

Comment:
We added a `useArrowKeyFocusManager` hook, let's use that instead. It's a bit simpler.